ironfish-cli/src/commands/airdrop/airdrop.ts

```diff
@@ -0,0 +1,48 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { CliUx, Flags } from '@oclif/core'
+import fs from 'fs/promises'
+import { IronfishCommand } from '../../command'
+import { LocalFlags } from '../../flags'
+import { watchTransaction } from '../../utils/transaction'
+
+export class Airdrop extends IronfishCommand {
+  static description = `Drop coins to testnet participants`
+  static aliases = ['airdrop:airdrop']
+  static hidden = true
+  static flags = {
+    ...LocalFlags,
+    posted: Flags.string({
+      required: true,
+      default: 'posted_transactions.txt',
+      description: 'New line separated serialized airdrop transactions',
+    }),
+  }
+  async start(): Promise<void> {
+    const { flags } = await this.parse(Airdrop)
+    let lineNum = 0
+    const fileContent = await fs.readFile(flags.posted, 'utf-8')
+    const lines = fileContent.split(/[\r\n]+/)
+    const client = await this.sdk.connectRpc()
+    for (const line of lines) {
+      lineNum++
+      CliUx.ux.action.start(`Adding transaction #${lineNum}`)
+      const response = await client.addTransaction({ transaction: line.trim() })
+      CliUx.ux.action.stop()
+      if (response.content.accepted) {
+        this.logger.info(`Added ${response.content.hash} transaction (#${lineNum})`)
+        await watchTransaction({
```

Comment:
Is it important to wait all the way until a transaction is confirmed before sending the next? They should be independent, but I could see not wanting to flood the network + mempool with a bunch of airdrop transactions.

If you do want to release them one at a time, it could be a bit quicker to wait for UNCONFIRMED. That way they'll be sent out in back-to-back blocks rather than separated by CONFIRMATION_RANGE # of blocks.

Also, do you have a plan to watch for transactions that expire without getting added to a block?

Reply:
Yea I would be fine sending them all at once, for some reason I thought Jason wanted to do individual in a row. As for the expiration I actually had a note about this, I was wondering if we should just make these have really long expiration times?

Reply:
Current state is that I have confirmations set to 0 and we are waiting, open to remove the wait if that is desirable.